### PR TITLE
Setup circleci ADC

### PR DESCRIPTION
**Pull request by jerjou**
Also, move all tests to use the project given in the environment variable, using buckets named after the project.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+machine:
+  java:
+    version: oraclejdk8
```
Comment (lesv) [inline circle.yml]: openjdk8 ??

Comment (jerjou) [inline circle.yml]: Yeah, for some reason I thought we were using oracle. Are we using openjdk?

Comment (jerjou) [inline circle.yml]: Hm.. doesn't look like circle [has openjdk8](https://circleci.com/docs/1.0/build-image-precise/#java), so if we want java 8, we have to go with oracle.

**Comment by jerjou** [comment]
Okay - spanner samples were failing intermittently, but I think it's because both travis and circle were running at the same time and clobbering state. I'm going to merge and then turn off travis.

**Pull request merged**